lib/reply.js

```diff
@@ -330,8 +330,12 @@ function notFound (reply) {
     return
   }
 
-  reply.context = reply.context._404Context
-  reply.context.handler(reply.request, reply)
+  // Use a copy of the 404 context with the original context's onSend hooks
+  const context = Object.assign({}, reply.context._404Context)
```

Comment:
Can you use prototypical inheritance here (Object.create..)? So that we avoid a full copy, the context is big and assign is slow.